lib/active_merchant/billing/gateways/authorize_net.rb

```diff
@@ -176,13 +176,40 @@ def credit(amount, payment, options = {})
         end
       end
 
-      def verify(credit_card, options = {})
+      def verify(payment_method, options = {})
+        amount = amount_for_verify(payment_method, options)
+        return amount if amount.is_a?(Response)
+
         MultiResponse.run(:use_first_response) do |r|
-          r.process { authorize(100, credit_card, options) }
-          r.process(:ignore_result) { void(r.authorization, options) }
+          r.process { authorize(amount, payment_method, options) }
+          r.process(:ignore_result) { void(r.authorization, options) } unless amount == 0
         end
       end
 
+      def amount_for_verify(payment_method, options)
+        if options[:verify_amount].present?
+          options[:verify_amount] = options[:verify_amount].to_i
```

Comment:
@jherreraa  I don't think we want to remove the `is_a?` check for Integer. In this new version, if the value of `:verify_amount` is a string, this line will conver it to a `0` which would proceed to attempt a 0 amount verify.